commands/analyze.go

```diff
@@ -39,109 +39,142 @@ func init() {
 
 func analyze(inDb string, configFile string) error {
 	res := resources.InitResources(configFile)
-	var toRunDirty []string
-	var toRun []string
+	var toRunDirty []database.RITADatabase
+	var toRun []database.RITADatabase
 
 	// Check to see if we want to run a full database or just one off the command line
 	if inDb == "" {
-		res.Log.Info("Running analysis against all databases")
-		toRunDirty = append(toRun, res.MetaDB.GetUnAnalyzedDatabases()...)
-	} else {
-		toRunDirty = append(toRun, inDb)
-	}
+		res.Log.Info("Running analysis against all unanalyzed databases")
+		unanalyzedDBs, err := res.DBIndex.GetUnanalyzedDatabases()
+		if err != nil {
+			errStr := fmt.Sprintf("Error: failed to list unanalyzed databases: %s", err.Error())
+			res.Log.Error(errStr)
+			return cli.NewExitError(errStr, 255)
+		}
 
-	// Check for problems
-	for _, possDB := range toRunDirty {
-		info, err := res.MetaDB.GetDBMetaInfo(possDB)
+		//ensure the databases have finished being imported
+		for i := range unanalyzedDBs {
+			if unanalyzedDBs[i].ImportFinished() {
+				toRunDirty = append(toRunDirty, unanalyzedDBs[i])
+			}
+		}
+	} else {
+		specifiedDB, err := res.DBIndex.GetDatabase(inDb)
 		if err != nil {
-			errStr := fmt.Sprintf("Error: %s not found.", possDB)
-			res.Log.Errorf(errStr)
-			fmt.Println(errStr)
-			continue
+			errStr := fmt.Sprintf("Error: database %s not found", inDb)
+			res.Log.Error(errStr)
+			return cli.NewExitError(errStr, 255)
 		}
-		if info.Analyzed {
-			errStr := fmt.Sprintf("Error: %s is already analyzed.", possDB)
-			res.Log.Errorf(errStr)
-			fmt.Println(errStr)
-			continue
+		if specifiedDB.Analyzed() {
+			errStr := fmt.Sprintf("Error: %s is already analyzed.", specifiedDB.Name())
+			res.Log.Error(errStr)
+			return cli.NewExitError(errStr, 255)
+		}
+		if !specifiedDB.ImportFinished() {
+			errStr := fmt.Sprintf("Error: %s hasn't finished importing.", specifiedDB.Name())
+			res.Log.Error(errStr)
+			return cli.NewExitError(errStr, 255)
 		}
-		semVer, err := semver.ParseTolerant(info.ImportVersion)
+		toRunDirty = append(toRunDirty, specifiedDB)
+	}
+
+	// Check for version problems
+	for i := range toRunDirty {
+		compatible, err := toRunDirty[i].CompatibleImportVersion(res.Config.R.Version)
```

Comment:
Checking for database compatibility has been moved into RITADatabase.CompatibleImportVersion